test/txstart-test.js

```diff
@@ -0,0 +1,228 @@
+
+    await assert.rejects(node.chain.add(block, VERIFY_NONE),
+      {reason: 'no-tx-allowed-yet'});
+    assert(node.chain.hasInvalid(block));
+  });
+
+  it('should reject non-empty block before txStart', async () => {
+    const tx1 = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+    tx1.locktime = node.chain.height + 1;
+
+    const tx2 = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+
+    const block = new Block();
+    block.txs.push(tx1);
+    block.txs.push(tx2);
+    block.prevBlock = node.chain.tip.hash;
+    block.time = node.chain.tip.time + 2;
+    block.bits = await node.chain.getTarget(block.time, node.chain.tip);
+
+    await assert.rejects(node.chain.add(block, VERIFY_NONE),
+      {reason: 'no-tx-allowed-yet'});
+    assert(node.chain.hasInvalid(block));
+  });
+
+  it('should accept empty block before txStart', async () => {
+    // Create an address that takes literally nothing to spend
+    const addr = Address.fromScript(new Script());
+
+    const tx1 = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+    tx1.outputs[0].address = addr;
+    tx1.locktime = node.chain.height + 1;
+
+    const block = new Block();
+    block.txs.push(tx1);
+    block.prevBlock = node.chain.tip.hash;
+    block.time = node.chain.tip.time + 3;
+    block.bits = await node.chain.getTarget(block.time, node.chain.tip);
+
+    assert(await node.chain.add(block, VERIFY_NONE));
+
+    // Spend this output later
+    utxo = block.txs[0].hash();
+  });
+
+  it('should add blocks until txStart is reached', async() => {
+    for (let i = node.chain.height; i < node.network.txStart - 1; i++) {
+      const block = await node.miner.mineBlock();
+      assert(await node.chain.add(block));
+    }
+    // The NEXT block is the txStart height
+    assert.strictEqual(node.chain.height + 1, node.network.txStart);
+  });
+
+  it('should allow TX in mempool at txStart height', async () => {
+    lastTX = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+    lastTX.inputs[0].prevout.hash = utxo;
+    lastTX.inputs[0].prevout.index = 0;
+    lastTX.inputs[0].witness.items[0] = Buffer.from([1, 1]);  // true
+    lastTX.inputs[0].witness.items[1] = Buffer.alloc(0);      // empty script
+
+    await node.mempool.addTX(lastTX);
+    assert.strictEqual(node.mempool.map.size, 1);
+    assert(node.mempool.has(lastTX.hash()));
+  });
+
+  it('should allow claim in mempool at txStart height', async () => {
+    const claim = await wallet.fakeClaim('cloudflare');
+
+    try {
+      ownership.ignore = true;
+      await node.mempool.addClaim(claim);
+    } finally {
+      ownership.ignore = false;
+    }
+    assert.strictEqual(node.mempool.claims.size, 1);
+    assert(node.mempool.hasClaim(claim.hash()));
+  });
+
+  it('should allow airdrop in mempool at txStart height', async () => {
```

Comment:
Since this test is specifically testing in context to a blockheight, it could use an assertion that the height is as expected. It requires context from previous tests to be certain that it is the correct height. This will also prevent regressions if any tests are changed in this file.